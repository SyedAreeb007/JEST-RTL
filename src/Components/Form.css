/* Form.css */

.container {
    margin-top: 800px;
    max-width: 500px;
    margin: 50px auto; /* Added some top margin */
    padding: 20px;
    border: 1px solid #ccc;
    border-radius: 10px;
    background-color: #f8f9fa;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); /* Added a subtle box shadow */
}

h2 {
    margin-bottom: 20px;
    text-align: center;
}

.form-group {
    margin-bottom: 20px;
}

label {
    font-weight: bold;
}

input[type="text"],
input[type="email"],
textarea {
    width: 100%;
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 5px;
    font-size: 1rem;
}

textarea {
    resize: vertical;
    min-height: 150px;
}

.btn-submit {
    background-color: #007bff;
    color: #fff;
    border: none;
    border-radius: 5px;
    padding: 10px 20px;
    font-size: 1rem;
    cursor: pointer;
    transition: background-color 0.3s;
    display: block;
    width: 100%;
}

.btn-submit:hover {
    background-color: #0056b3;
}
