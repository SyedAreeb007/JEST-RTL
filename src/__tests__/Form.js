import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Form from '../Components/Form';

describe('Form Component', () => {
  test('renders the form with all fields', () => {
    const { asFragment } = render(<Form />);
    expect(screen.getByLabelText(/name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/email/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/message/i)).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();

  });

  test('submits the form with the correct data', () => {
    render(<Form />);
    
    const nameInput = screen.getByLabelText(/name/i);
    const emailInput = screen.getByLabelText(/email/i);
    const messageInput = screen.getByLabelText(/message/i);
    const submitButton = screen.getByText(/submit/i);

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(emailInput, { target: { value: 'john@example.com' } });
    fireEvent.change(messageInput, { target: { value: 'Hello!' } });

    fireEvent.click(submitButton);

    expect(nameInput.value).toBe('John Doe');
    expect(emailInput.value).toBe('john@example.com');
    expect(messageInput.value).toBe('Hello!');
  });
});
