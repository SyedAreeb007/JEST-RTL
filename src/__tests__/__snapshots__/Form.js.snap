// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders the form with all fields 1`] = `[Function]`;
